docs/usage/4-request-data.md

```diff
@@ -0,0 +1,117 @@
+# Request Data
+
+## Request body
+
+The body of HTTP requests can be accessed using the special `data` parameter in a handler function.
+
+```py
+--8<-- "examples/request_data/request_data_1.py"
+```
+
+
+The type of `data` an be any supported type, including
+
+- [dataclasses](https://docs.python.org/3/library/dataclasses.html)
+- [`TypedDict`s][typing.TypedDict]
+- Pydantic models
+- Arbitrary stdlib types
+- Typed supported via [plugins](/usage/10-plugins/0-plugins-intro.md)
+
+
+```py
+--8<-- "examples/request_data/request_data_2.py"
+```
+
+
+## Validation and customizing OpenAPI documentation
+
+With the help of [Body][starlite.params.Body], you have fine-grained control over the validation
+of the request body, and can also customize the OpenAPI documentation:
+
+```py
+--8<-- "examples/request_data/request_data_3.py"
+```
+
+
+## Specifying a content-type
+
+By default, Starlite will try to parse the request body as JSON. While this may be desired
+in most cases, you might want to specify a different type. You can do so by passing a
+[RequestEncodingType][starlite.enums.RequestEncodingType] to `Body`. This will also
+help to generate the correct media-type in the OpenAPI schema.
+
+### URL Encoded Form Data
+
+To access data sent as [url-encoded form data](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/POST),
+i.e. `application/x-www-form-urlencoded` Content-Type header, use [`Body`][starlite.params.Body] and specify
+[`RequestEncodingType.URL_ENCODED`][starlite.enums.RequestEncodingType] as the `media_type`:
+
+```py
+--8<-- "examples/request_data/request_data_4.py"
+```
+
+
+
+!!! info
+    URL encoded data is inherently less versatile than JSON data - for example, it cannot handle complex
+    dictionaries and deeply nested data. It should only be used for simple data structures.
+
+### MultiPart Form Data
+
+You can access data uploaded using a request with a [`multipart/form-data`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/POST)
+Content-Type header by specifying it in the [`Body`][starlite.params.Body] function:
+
+```py
+--8<-- "examples/request_data/request_data_5.py"
+```
+
+
+## File uploads
```

Comment:
Why not have this section in a separate MD file?

Reply:
Because it's a much easier read to have all this within one file with proper sections and subsection headers. You can still navigate to subsections via the TOC if you want, so it's functionally the same as separate files, only difference is that it makes for a better continuous read, and you have an immediate overview of all sections and subsections.
 
There's really no advantage to having this as separate files.